app/services/git_hub_authenticator.rb

```diff
@@ -0,0 +1,57 @@
+class GitHubAuthenticator
+  def initialize(auth, current_user = nil)
+    @auth = auth
+    @current_user = current_user
+  end
+
+  def self.authenticate(auth, current_user = nil)
+    new(auth, current_user).authenticate
+  end
+
+  def authenticate
+    if user = find_user
+      user.update_attributes(github_params)
```

Comment:
We should use `update_attributes!` here as we are not checking return value of this statement.